internal/settings/gandi.go

```diff
@@ -0,0 +1,199 @@
+		dnsLookup: !noDNSLookup,
+		key:       extraSettings.Key,
+		ttl:       extraSettings.TTL,
+	}
+	if err := g.isValid(); err != nil {
+		return nil, err
+	}
+	return d, nil
+}
+
+func (g *gandi) isValid() error {
+	if len(g.key) == 0 {
+		return ErrEmptyKey
+	}
+	return nil
+}
+
+func (g *gandi) String() string {
+	return toString(g.domain, g.host, constants.GANDI, g.ipVersion)
+}
+
+func (g *gandi) Domain() string {
+	return g.domain
+}
+
+func (g *gandi) Host() string {
+	return g.host
+}
+
+func (g *gandi) DNSLookup() bool {
+	return g.dnsLookup
+}
+
+func (g *gandi) IPVersion() models.IPVersion {
+	return g.ipVersion
+}
+
+func (g *gandi) BuildDomainName() string {
+	return buildDomainName(g.host, g.domain)
+}
+
+func (g *gandi) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    models.HTML(fmt.Sprintf("<a href=\"http://%s\">%s</a>", g.BuildDomainName(), g.BuildDomainName())),
+		Host:      models.HTML(g.Host()),
+		Provider:  "<a href=\"https://www.gandi.net/\">gandi</a>",
+		IPVersion: models.HTML(g.ipVersion),
+	}
+}
+
+func (g *gandi) setHeaders(request *http.Request) {
+	setUserAgent(request)
+	setContentType(request, "application/json")
+	setAccept(request, "application/json")
+	request.Header.Set("X-Api-Key", g.key)
+}
+
+func (g *gandi) getRecordIP(ctx context.Context, recordType string, client *http.Client) (
+	recordIP string, err error) {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "dns.api.gandi.net",
+		Path:   fmt.Sprintf("/api/v5/domains/%s/records/%s/%s", g.domain, g.host, recordType),
+	}
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return "", err
+	}
+	g.setHeaders(request)
+
+	response, err := client.Do(request)
+	if err != nil {
+		return "", err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("%w: %d: %s",
+			ErrBadHTTPStatus, response.StatusCode, bodyToSingleLine(response.Body))
+	}
+
+	decoder := json.NewDecoder(response.Body)
+	var result struct {
+		Type   string   `json:"rrset_type"`
+		TTL    int      `json:"rrset_ttl"`
+		Name   string   `json:"rrset_name"`
+		Href   string   `json:"rrset_href"`
+		Values []string `json:"rrset_values"`
+	}
+	if err = decoder.Decode(&result); err != nil {
+		return "", fmt.Errorf("%w: %s", ErrUnmarshalResponse, err)
+	}
+	if len(result.Values) == 0 {
+		return "", ErrNoResultReceived
+	}
+	return result.Values[0], nil
+}
+
+func (g *gandi) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP net.IP, err error) {
+	recordType := A
+	if ip.To4() == nil { // IPv6
+		recordType = AAAA
+	}
+
+	recordIP, err := g.getRecordIP(ctx, recordType, client)
+	if err != nil && err != ErrNoResultReceived { // if no ip was defined before, let's proceed with the update
+		return nil, fmt.Errorf("%s: %w", ErrGetRecordIP, err)
+	}
+
+	oldIP := net.ParseIP(recordIP)
+	if ip.Equal(oldIP) { // success, nothing to change
+		return ip, nil
+	}
```

Comment:
Sorry for the wasted effort, but this can be removed as it's already doing a DNS resolution against your domain to check if the IP changed in some other part of the code. **Except** if the domain can be proxied (which makes the DNS resolution mechanism check not working), but for now I think only Cloudflare offers such service.

```suggestion
```

Reply:
Here I was trying to check if the IP on the registrar is correct or not in order to decide if we want to update the IP or not. Indeed it might not be worth checking, it duplicates the number of calls for no gain